mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -215,6 +216,10 @@ def _startup(self, bus, skill_id=""):
             self.skill_id = skill_id or basename(self.root_dir)
             self.intent_service.set_id(self.skill_id)
             self.event_scheduler.set_id(self.skill_id)
+            self.enclosure.set_id(self.skill_id)
+
+            # initialize anything that depends on skill_id
+            self.log = LOG.create_logger(self.skill_id)
```

Comment:
Is it desirable to create a new logger instead of using LOG directly? I know this is how Mycroft did it, but I never really understood why

Reply:
changes the name of the logger internally and help filtering logs